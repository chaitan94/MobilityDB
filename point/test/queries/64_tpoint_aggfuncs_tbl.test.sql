-------------------------------------------------------------------------------
set parallel_tuple_cost=0;
set parallel_setup_cost=0;
set force_parallel_mode=regress;

-------------------------------------------------------------------------------
-- Extent aggregate function
-------------------------------------------------------------------------------

select extent(inst) from tbl_tgeompointinst;
select extent(inst) from tbl_tgeogpointinst;
select extent(ti) from tbl_tgeompointi;
select extent(ti) from tbl_tgeogpointi;
select extent(seq) from tbl_tgeompointseq;
select extent(seq) from tbl_tgeogpointseq;
select extent(ts) from tbl_tgeompoints;
select extent(ts) from tbl_tgeogpoints;
select extent(temp) from tbl_tgeompoint;
select extent(temp) from tbl_tgeogpoint;

select extent(inst) from tbl_tgeompoint3Dinst;
select extent(inst) from tbl_tgeogpoint3Dinst;
select extent(ti) from tbl_tgeompoint3Di;
select extent(ti) from tbl_tgeogpoint3Di;
select extent(seq) from tbl_tgeompoint3Dseq;
select extent(seq) from tbl_tgeogpoint3Dseq;
select extent(ts) from tbl_tgeompoint3Ds;
select extent(ts) from tbl_tgeogpoint3Ds;
select extent(temp) from tbl_tgeompoint3D;
select extent(temp) from tbl_tgeogpoint3D;

-------------------------------------------------------------------------------

SELECT numInstants(tcentroid(inst)) FROM tbl_tgeompointinst;
SELECT numInstants(tcount(inst)) FROM tbl_tgeompointinst;
SELECT k%10, numInstants(tcentroid(inst)) FROM tbl_tgeompointinst GROUP BY k%10 ORDER BY k%10;
SELECT k%10, numInstants(tcount(inst)) FROM tbl_tgeompointinst GROUP BY k%10 ORDER BY k%10;

SELECT numInstants(tcentroid(ti)) FROM tbl_tgeompointi;
SELECT numInstants(tcount(ti)) FROM tbl_tgeompointi;
SELECT k%10, numInstants(tcentroid(ti)) FROM tbl_tgeompointi GROUP BY k%10 ORDER BY k%10;
SELECT k%10, numInstants(tcount(ti)) FROM tbl_tgeompointi GROUP BY k%10 ORDER BY k%10;

SELECT numSequences(tcentroid(seq)) FROM tbl_tgeompointseq;
SELECT numSequences(tcount(seq)) FROM tbl_tgeompointseq;
SELECT k%10, numSequences(tcentroid(seq)) FROM tbl_tgeompointseq GROUP BY k%10 ORDER BY k%10;
SELECT k%10, numSequences(tcount(seq)) FROM tbl_tgeompointseq GROUP BY k%10 ORDER BY k%10;

SELECT numSequences(tcentroid(ts)) FROM tbl_tgeompoints;
SELECT numSequences(tcount(ts)) FROM tbl_tgeompoints;
SELECT k%10, numSequences(tcentroid(ts)) FROM tbl_tgeompoints GROUP BY k%10 ORDER BY k%10;
SELECT k%10, numSequences(tcount(ts)) FROM tbl_tgeompoints GROUP BY k%10 ORDER BY k%10;

SELECT numInstants(tcentroid(inst)) FROM tbl_tgeompoint3Dinst;
SELECT numInstants(tcount(inst)) FROM tbl_tgeompoint3Dinst;
SELECT k%10, numInstants(tcentroid(inst)) FROM tbl_tgeompoint3Dinst GROUP BY k%10 ORDER BY k%10;
SELECT k%10, numInstants(tcount(inst)) FROM tbl_tgeompoint3Dinst GROUP BY k%10 ORDER BY k%10;

SELECT numInstants(tcentroid(ti)) FROM tbl_tgeompoint3Di;
SELECT numInstants(tcount(ti)) FROM tbl_tgeompoint3Di;
SELECT k%10, numInstants(tcentroid(ti)) FROM tbl_tgeompoint3Di GROUP BY k%10 ORDER BY k%10;
SELECT k%10, numInstants(tcount(ti)) FROM tbl_tgeompoint3Di GROUP BY k%10 ORDER BY k%10;

SELECT numSequences(tcentroid(seq)) FROM tbl_tgeompoint3Dseq;
SELECT numSequences(tcount(seq)) FROM tbl_tgeompoint3Dseq;
SELECT k%10, numSequences(tcentroid(seq)) FROM tbl_tgeompoint3Dseq GROUP BY k%10 ORDER BY k%10;
SELECT k%10, numSequences(tcount(seq)) FROM tbl_tgeompoint3Dseq GROUP BY k%10 ORDER BY k%10;

SELECT numSequences(tcentroid(ts)) FROM tbl_tgeompoint3Ds;
SELECT numSequences(tcount(ts)) FROM tbl_tgeompoint3Ds;
SELECT k%10, numSequences(tcentroid(ts)) FROM tbl_tgeompoint3Ds GROUP BY k%10 ORDER BY k%10;
SELECT k%10, numSequences(tcount(ts)) FROM tbl_tgeompoint3Ds GROUP BY k%10 ORDER BY k%10;

-------------------------------------------------------------------------------

SELECT numInstants(tcount(inst)) FROM tbl_tgeogpointinst;
SELECT k%10, numInstants(tcount(inst)) FROM tbl_tgeogpointinst GROUP BY k%10 ORDER BY k%10;

SELECT numInstants(tcount(ti)) FROM tbl_tgeogpointi;
SELECT k%10, numInstants(tcount(ti)) FROM tbl_tgeogpointi GROUP BY k%10 ORDER BY k%10;

SELECT numSequences(tcount(seq)) FROM tbl_tgeogpointseq;
SELECT k%10, numSequences(tcount(seq)) FROM tbl_tgeogpointseq GROUP BY k%10 ORDER BY k%10;

SELECT numSequences(tcount(ts)) FROM tbl_tgeogpoints;
SELECT k%10, numSequences(tcount(ts)) FROM tbl_tgeogpoints GROUP BY k%10 ORDER BY k%10;

-------------------------------------------------------------------------------

set parallel_tuple_cost=100;
set parallel_setup_cost=100;
set force_parallel_mode=off;

-------------------------------------------------------------------------------
