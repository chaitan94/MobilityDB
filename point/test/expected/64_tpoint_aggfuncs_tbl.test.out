set parallel_tuple_cost=0;
SET
set parallel_setup_cost=0;
SET
set force_parallel_mode=regress;
SET
select extent(inst) from tbl_tgeompointinst;
                                              extent                                              
--------------------------------------------------------------------------------------------------
 STBOX T((1.8961993,0.7555033,2001-01-01 22:05:00+00),(99.864342,99.5075,2001-12-11 14:43:00+00))
(1 row)

select extent(inst) from tbl_tgeogpointinst;
                                                              extent                                                               
-----------------------------------------------------------------------------------------------------------------------------------
 GEODSTBOX ZT((0.30129021,-0.12238938,0.5739544,2001-01-04 09:31:00+00),(0.81489998,0.37733009,0.94804142,2001-12-30 22:05:00+00))
(1 row)

select extent(ti) from tbl_tgeompointi;
                                                extent                                                
------------------------------------------------------------------------------------------------------
 STBOX T((0.09850096,0.86371142,2001-01-01 12:09:00+00),(99.882332,99.407364,2001-12-23 15:39:00+00))
(1 row)

select extent(ti) from tbl_tgeogpointi;
                                                              extent                                                               
-----------------------------------------------------------------------------------------------------------------------------------
 GEODSTBOX ZT((0.26595744,-0.13913035,0.57390403,2001-01-10 21:37:00+00),(0.81890941,0.41401273,0.9504269,2001-12-29 00:22:00+00))
(1 row)

select extent(seq) from tbl_tgeompointseq;
                                                extent                                                
------------------------------------------------------------------------------------------------------
 STBOX T((0.31485471,0.10301345,2001-01-08 09:36:00+00),(99.932587,99.580742,2001-12-25 04:46:00+00))
(1 row)

select extent(seq) from tbl_tgeogpointseq;
                                                               extent                                                               
------------------------------------------------------------------------------------------------------------------------------------
 GEODSTBOX ZT((0.26864859,-0.12480921,0.57369322,2001-01-02 19:00:00+00),(0.81903058,0.41619259,0.95074242,2001-12-19 18:07:00+00))
(1 row)

select extent(ts) from tbl_tgeompoints;
                                                extent                                                
------------------------------------------------------------------------------------------------------
 STBOX T((0.01012939,0.052811205,2001-01-01 10:09:00+00),(99.953613,99.93779,2001-12-30 10:07:00+00))
(1 row)

select extent(ts) from tbl_tgeogpoints;
                                                             extent                                                              
---------------------------------------------------------------------------------------------------------------------------------
 GEODSTBOX ZT((0.2699886,-0.13447556,0.5735808,2001-01-01 20:01:00+00),(0.81845045,0.4233391,0.95098066,2001-12-22 20:33:00+00))
(1 row)

select extent(temp) from tbl_tgeompoint;
                                              extent                                               
---------------------------------------------------------------------------------------------------
 STBOX T((0.09850096,0.5681029,2001-01-01 10:09:00+00),(99.93586,99.93779,2001-12-30 10:07:00+00))
(1 row)

select extent(temp) from tbl_tgeogpoint;
                                                              extent                                                               
-----------------------------------------------------------------------------------------------------------------------------------
 GEODSTBOX ZT((0.27862182,-0.13447556,0.57363325,2001-01-04 09:31:00+00),(0.81903058,0.40801156,0.9502285,2001-12-17 18:56:00+00))
(1 row)

select extent(inst) from tbl_tgeompoint3Dinst;
                                                           extent                                                           
----------------------------------------------------------------------------------------------------------------------------
 STBOX ZT((0.096990302,0.91515237,1.2832818,2001-01-01 05:13:00+00),(95.650017,99.573982,98.010948,2001-12-24 21:44:00+00))
(1 row)

select extent(inst) from tbl_tgeogpoint3Dinst;
                                                              extent                                                               
-----------------------------------------------------------------------------------------------------------------------------------
 GEODSTBOX ZT((0.27987459,-0.10442724,0.5743441,2001-01-04 03:22:00+00),(0.81626362,0.36257252,0.94793653,2001-12-28 09:03:00+00))
(1 row)

select extent(ti) from tbl_tgeompoint3Di;
                                                          extent                                                           
---------------------------------------------------------------------------------------------------------------------------
 STBOX ZT((0.37831783,0.73828566,0.43434849,2001-01-09 23:49:00+00),(99.873901,99.91011,99.830681,2001-12-28 14:07:00+00))
(1 row)

select extent(ti) from tbl_tgeogpoint3Di;
                                                             extent                                                              
---------------------------------------------------------------------------------------------------------------------------------
 GEODSTBOX ZT((0.27067214,-0.116604,0.57441998,2001-01-06 01:21:00+00),(0.81704789,0.4274379,0.95095807,2001-12-18 06:24:00+00))
(1 row)

select extent(seq) from tbl_tgeompoint3Dseq;
                                                           extent                                                           
----------------------------------------------------------------------------------------------------------------------------
 STBOX ZT((0.016309181,0.19594561,0.21574804,2001-01-02 01:22:00+00),(99.934143,99.71315,99.870621,2001-12-29 08:08:00+00))
(1 row)

select extent(seq) from tbl_tgeogpoint3Dseq;
                                                              extent                                                               
-----------------------------------------------------------------------------------------------------------------------------------
 GEODSTBOX ZT((0.27951702,-0.13062911,0.5736379,2001-01-01 05:04:00+00),(0.81496489,0.40756309,0.95081085,2001-12-14 14:34:00+00))
(1 row)

select extent(ts) from tbl_tgeompoint3Ds;
                                                           extent                                                            
-----------------------------------------------------------------------------------------------------------------------------
 STBOX ZT((0.013936218,0.079045951,0.053835984,2001-01-01 08:42:00+00),(99.99707,99.940804,99.98185,2001-12-30 14:00:00+00))
(1 row)

select extent(ts) from tbl_tgeogpoint3Ds;
                                                              extent                                                              
----------------------------------------------------------------------------------------------------------------------------------
 GEODSTBOX ZT((0.26368076,-0.13838577,0.57409668,2001-01-02 23:15:00+00),(0.81495869,0.4136003,0.9510355,2001-12-30 19:51:00+00))
(1 row)

select extent(temp) from tbl_tgeompoint3D;
                                                         extent                                                          
-------------------------------------------------------------------------------------------------------------------------
 STBOX ZT((0.028999999,0.1534508,0.19856402,2001-01-01 05:13:00+00),(99.99707,99.91011,99.98185,2001-12-30 14:00:00+00))
(1 row)

select extent(temp) from tbl_tgeogpoint3D;
                                                              extent                                                              
----------------------------------------------------------------------------------------------------------------------------------
 GEODSTBOX ZT((0.27067214,-0.1366867,0.5743441,2001-01-04 03:45:00+00),(0.81704789,0.41072083,0.95049638,2001-12-23 08:21:00+00))
(1 row)

SELECT numInstants(tcentroid(inst)) FROM tbl_tgeompointinst;
 numinstants 
-------------
          96
(1 row)

SELECT numInstants(tcount(inst)) FROM tbl_tgeompointinst;
 numinstants 
-------------
          96
(1 row)

SELECT k%10, numInstants(tcentroid(inst)) FROM tbl_tgeompointinst GROUP BY k%10 ORDER BY k%10;
 ?column? | numinstants 
----------+-------------
        0 |          10
        1 |          10
        2 |          10
        3 |          10
        4 |          10
        5 |          10
        6 |          10
        7 |          10
        8 |          10
        9 |          10
(10 rows)

SELECT k%10, numInstants(tcount(inst)) FROM tbl_tgeompointinst GROUP BY k%10 ORDER BY k%10;
 ?column? | numinstants 
----------+-------------
        0 |          10
        1 |          10
        2 |          10
        3 |          10
        4 |          10
        5 |          10
        6 |          10
        7 |          10
        8 |          10
        9 |          10
(10 rows)

SELECT numInstants(tcentroid(ti)) FROM tbl_tgeompointi;
 numinstants 
-------------
         454
(1 row)

SELECT numInstants(tcount(ti)) FROM tbl_tgeompointi;
 numinstants 
-------------
         454
(1 row)

SELECT k%10, numInstants(tcentroid(ti)) FROM tbl_tgeompointi GROUP BY k%10 ORDER BY k%10;
 ?column? | numinstants 
----------+-------------
        0 |          59
        1 |          54
        2 |          39
        3 |          44
        4 |          56
        5 |          43
        6 |          52
        7 |          50
        8 |          41
        9 |          41
(10 rows)

SELECT k%10, numInstants(tcount(ti)) FROM tbl_tgeompointi GROUP BY k%10 ORDER BY k%10;
 ?column? | numinstants 
----------+-------------
        0 |          59
        1 |          54
        2 |          39
        3 |          44
        4 |          56
        5 |          43
        6 |          52
        7 |          50
        8 |          41
        9 |          41
(10 rows)

SELECT numSequences(tcentroid(seq)) FROM tbl_tgeompointseq;
 numsequences 
--------------
           99
(1 row)

SELECT numSequences(tcount(seq)) FROM tbl_tgeompointseq;
 numsequences 
--------------
           95
(1 row)

SELECT k%10, numSequences(tcentroid(seq)) FROM tbl_tgeompointseq GROUP BY k%10 ORDER BY k%10;
 ?column? | numsequences 
----------+--------------
        0 |           10
        1 |           10
        2 |           10
        3 |           11
        4 |           10
        5 |           10
        6 |           10
        7 |           10
        8 |           10
        9 |           10
(10 rows)

SELECT k%10, numSequences(tcount(seq)) FROM tbl_tgeompointseq GROUP BY k%10 ORDER BY k%10;
 ?column? | numsequences 
----------+--------------
        0 |           10
        1 |           10
        2 |           10
        3 |           10
        4 |           10
        5 |           10
        6 |           10
        7 |           10
        8 |           10
        9 |           10
(10 rows)

SELECT numSequences(tcentroid(ts)) FROM tbl_tgeompoints;
 numsequences 
--------------
          474
(1 row)

SELECT numSequences(tcount(ts)) FROM tbl_tgeompoints;
 numsequences 
--------------
          455
(1 row)

SELECT k%10, numSequences(tcentroid(ts)) FROM tbl_tgeompoints GROUP BY k%10 ORDER BY k%10;
 ?column? | numsequences 
----------+--------------
        0 |           35
        1 |           56
        2 |           37
        3 |           51
        4 |           46
        5 |           54
        6 |           43
        7 |           51
        8 |           44
        9 |           64
(10 rows)

SELECT k%10, numSequences(tcount(ts)) FROM tbl_tgeompoints GROUP BY k%10 ORDER BY k%10;
 ?column? | numsequences 
----------+--------------
        0 |           35
        1 |           56
        2 |           37
        3 |           51
        4 |           46
        5 |           54
        6 |           43
        7 |           51
        8 |           44
        9 |           64
(10 rows)

SELECT numInstants(tcentroid(inst)) FROM tbl_tgeompoint3Dinst;
 numinstants 
-------------
          96
(1 row)

SELECT numInstants(tcount(inst)) FROM tbl_tgeompoint3Dinst;
 numinstants 
-------------
          96
(1 row)

SELECT k%10, numInstants(tcentroid(inst)) FROM tbl_tgeompoint3Dinst GROUP BY k%10 ORDER BY k%10;
 ?column? | numinstants 
----------+-------------
        0 |          10
        1 |          10
        2 |          10
        3 |          10
        4 |          10
        5 |          10
        6 |          10
        7 |          10
        8 |          10
        9 |          10
(10 rows)

SELECT k%10, numInstants(tcount(inst)) FROM tbl_tgeompoint3Dinst GROUP BY k%10 ORDER BY k%10;
 ?column? | numinstants 
----------+-------------
        0 |          10
        1 |          10
        2 |          10
        3 |          10
        4 |          10
        5 |          10
        6 |          10
        7 |          10
        8 |          10
        9 |          10
(10 rows)

SELECT numInstants(tcentroid(ti)) FROM tbl_tgeompoint3Di;
 numinstants 
-------------
         495
(1 row)

SELECT numInstants(tcount(ti)) FROM tbl_tgeompoint3Di;
 numinstants 
-------------
         495
(1 row)

SELECT k%10, numInstants(tcentroid(ti)) FROM tbl_tgeompoint3Di GROUP BY k%10 ORDER BY k%10;
 ?column? | numinstants 
----------+-------------
        0 |          47
        1 |          53
        2 |          62
        3 |          63
        4 |          51
        5 |          52
        6 |          52
        7 |          51
        8 |          39
        9 |          58
(10 rows)

SELECT k%10, numInstants(tcount(ti)) FROM tbl_tgeompoint3Di GROUP BY k%10 ORDER BY k%10;
 ?column? | numinstants 
----------+-------------
        0 |          47
        1 |          53
        2 |          62
        3 |          63
        4 |          51
        5 |          52
        6 |          52
        7 |          51
        8 |          39
        9 |          58
(10 rows)

SELECT numSequences(tcentroid(seq)) FROM tbl_tgeompoint3Dseq;
 numsequences 
--------------
           99
(1 row)

SELECT numSequences(tcount(seq)) FROM tbl_tgeompoint3Dseq;
 numsequences 
--------------
           96
(1 row)

SELECT k%10, numSequences(tcentroid(seq)) FROM tbl_tgeompoint3Dseq GROUP BY k%10 ORDER BY k%10;
 ?column? | numsequences 
----------+--------------
        0 |           10
        1 |           10
        2 |           10
        3 |           10
        4 |           10
        5 |           10
        6 |           10
        7 |           10
        8 |           10
        9 |           10
(10 rows)

SELECT k%10, numSequences(tcount(seq)) FROM tbl_tgeompoint3Dseq GROUP BY k%10 ORDER BY k%10;
 ?column? | numsequences 
----------+--------------
        0 |           10
        1 |           10
        2 |           10
        3 |           10
        4 |           10
        5 |           10
        6 |           10
        7 |           10
        8 |           10
        9 |           10
(10 rows)

SELECT numSequences(tcentroid(ts)) FROM tbl_tgeompoint3Ds;
 numsequences 
--------------
          522
(1 row)

SELECT numSequences(tcount(ts)) FROM tbl_tgeompoint3Ds;
 numsequences 
--------------
          478
(1 row)

SELECT k%10, numSequences(tcentroid(ts)) FROM tbl_tgeompoint3Ds GROUP BY k%10 ORDER BY k%10;
 ?column? | numsequences 
----------+--------------
        0 |           54
        1 |           41
        2 |           65
        3 |           57
        4 |           48
        5 |           27
        6 |           41
        7 |           62
        8 |           48
        9 |           58
(10 rows)

SELECT k%10, numSequences(tcount(ts)) FROM tbl_tgeompoint3Ds GROUP BY k%10 ORDER BY k%10;
 ?column? | numsequences 
----------+--------------
        0 |           54
        1 |           41
        2 |           65
        3 |           57
        4 |           48
        5 |           27
        6 |           41
        7 |           62
        8 |           47
        9 |           58
(10 rows)

SELECT numInstants(tcount(inst)) FROM tbl_tgeogpointinst;
 numinstants 
-------------
          96
(1 row)

SELECT k%10, numInstants(tcount(inst)) FROM tbl_tgeogpointinst GROUP BY k%10 ORDER BY k%10;
 ?column? | numinstants 
----------+-------------
        0 |          10
        1 |          10
        2 |          10
        3 |          10
        4 |          10
        5 |          10
        6 |          10
        7 |          10
        8 |          10
        9 |          10
(10 rows)

SELECT numInstants(tcount(ti)) FROM tbl_tgeogpointi;
 numinstants 
-------------
         476
(1 row)

SELECT k%10, numInstants(tcount(ti)) FROM tbl_tgeogpointi GROUP BY k%10 ORDER BY k%10;
 ?column? | numinstants 
----------+-------------
        0 |          59
        1 |          66
        2 |          47
        3 |          50
        4 |          39
        5 |          42
        6 |          47
        7 |          47
        8 |          51
        9 |          49
(10 rows)

SELECT numSequences(tcount(seq)) FROM tbl_tgeogpointseq;
 numsequences 
--------------
           95
(1 row)

SELECT k%10, numSequences(tcount(seq)) FROM tbl_tgeogpointseq GROUP BY k%10 ORDER BY k%10;
 ?column? | numsequences 
----------+--------------
        0 |           10
        1 |           10
        2 |           10
        3 |           10
        4 |           10
        5 |           10
        6 |           10
        7 |           10
        8 |           10
        9 |           10
(10 rows)

SELECT numSequences(tcount(ts)) FROM tbl_tgeogpoints;
 numsequences 
--------------
          480
(1 row)

SELECT k%10, numSequences(tcount(ts)) FROM tbl_tgeogpoints GROUP BY k%10 ORDER BY k%10;
 ?column? | numsequences 
----------+--------------
        0 |           53
        1 |           50
        2 |           48
        3 |           44
        4 |           61
        5 |           54
        6 |           60
        7 |           55
        8 |           32
        9 |           43
(10 rows)

set parallel_tuple_cost=100;
SET
set parallel_setup_cost=100;
SET
set force_parallel_mode=off;
SET
